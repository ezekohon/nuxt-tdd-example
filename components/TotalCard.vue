<template>
  <div class="column">
    <div class="card">
      <header class="card-header">
        <p class="card-header-title has-text-danger">
          Total
        </p>
      </header>
      <footer class="card-footer">
        <p class="card-footer-item has-text-weight-bold">
          {{ `$${price}` }}
        </p>
      </footer>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    price: {
      type: Number,
      required: true
    }
  }
}
</script>

<style lang="scss" scoped>
</style>
