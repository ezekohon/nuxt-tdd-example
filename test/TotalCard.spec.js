import { mount } from '@vue/test-utils';
import TotalCard from '@/components/TotalCard'

let wrapper = null;

const props = {
    price: 21
};

beforeEach(() => {
    wrapper = mount(TotalCard, {
        propsData: {
            price: props.price
        }
    });
})

afterEach(() => {
    wrapper.destroy();
})

describe('TotalCard', () => {
    it('renders total price', () => {
        expect(wrapper.find('.card-footer-item').text()).toContain(props.price)
    })

    it('matches snapshot', () => {
        expect(wrapper.element).toMatchSnapshot()
    })
})