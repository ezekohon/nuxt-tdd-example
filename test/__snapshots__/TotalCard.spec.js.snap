// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalCard matches snapshot 1`] = `
<div
  class="column"
>
  <div
    class="card"
  >
    <header
      class="card-header"
    >
      <p
        class="card-header-title has-text-danger"
      >
        
        Total
      
      </p>
    </header>
     
    <footer
      class="card-footer"
    >
      <p
        class="card-footer-item has-text-weight-bold"
      >
        
        $21
      
      </p>
    </footer>
  </div>
</div>
`;
